common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -240,12 +240,26 @@ public static <T extends Appendable> T toHexString(T dst, byte[] src, int offset
      * given, or {@code -1} if the character is invalid.
      */
     public static int decodeHexNibble(final char c) {
-        assert HEX2B.length == (Character.MAX_VALUE + 1);
+        assert HEX2B.length == (Character.MAX_VALUE + 1) : "Invalid ASCII character";
         // Character.digit() is not used here, as it addresses a larger
         // set of characters (both ASCII and full-width latin letters).
         return HEX2B[c];
     }
 
+    /**
+     * Helper to decode half of a hexadecimal number from a string.
+     * @param b The ASCII character of the hexadecimal number to decode.
+     * Must be in the range {@code [0-9a-fA-F]}.
+     * @return The hexadecimal value represented in the ASCII character
+     * given, or {@code -1} if the character is invalid.
+     */
+    public static int decodeHexNibble(final byte b) {
+        assert HEX2B.length == (Character.MAX_VALUE + 1) : "Invalid ASCII character";
```

Comment:
hmm... I think this assert (And the one above) makes really no sense. We should assert when we create the `HEX2B` array.